### PR TITLE
docs: update user-event references

### PR DESCRIPTION
**Pull request by ph-fritsche**
Update all references to `user-event` preparing for the upcoming release.

This should be merged once the version is released on `latest`:
- [ ] `user-event@14` has been released.

### CODE REVIEW

**Comment by ph-fritsche** [comment]
Note: The `format` script produces broken block quotes in Markdown
```md
> bug fixes and new features. You can find the docs for `user-event@13.5.0`
> [here](../ecosystem-user-event.mdx).

becomes

> bug fixes and new features. You can find the docs for `user-event@13.5.0` >
> [here](../ecosystem-user-event.mdx).
```

File: docs/react-testing-library/migrate-from-enzyme.mdx [hunk 1]
```diff
@@ -330,10 +330,14 @@ input's "checked" property is properly set. Let's see how we might write a test
 for that case:
 
 ```jsx
-test('handles click correctly', () => {
+test('handles click correctly', async () => {
   render(<Checkbox />)
+  const user = userEvent.setup()
+
+  // You can also call this method directly on userEvent,
```
Comment (MatanBobi) [inline docs/react-testing-library/migrate-from-enzyme.mdx:337]: I'm not sure if that's even something we want to "share". If we believe that the recommended approach is to use the methods returned from `setup`, maybe this comment isn't helpful and can only cause bad practices. Wdyt?

**Review by MatanBobi — APPROVED** [review]
Looks good :)
Question about the formatting, it looks like a lot of things changed there, did we not run format before or maybe you're by any chance using a different prettier configs?

**Comment by ph-fritsche** [comment]
> did we not run format before or maybe you're by any chance using a different prettier configs?

I think I skipped `format` before.

I'm not sure I like what prettier does to the Markdown files. A lot of lines are much harder to comprehend with the line breaks at fixed width instead of at the end of subclauses.

File: docs/react-testing-library/migrate-from-enzyme.mdx (hunk 1 above)
Comment (ph-fritsche) [reply docs/react-testing-library/migrate-from-enzyme.mdx:337]: I think it eases the transition for people who are used to the previous APIs. That's why the [Direct APIs](https://testing-library.com/docs/user-event/setup#direct-apis) still exist.

If there is only a single API call, it makes no difference, as the Direct APIs call `.setup()` internally and then use the method from there.

If it helps or hurts to mention it in the example - I don't know.

**Comment by eps1lon** [comment]
Are the old links properly redirecting? For example, if an external site linked to a user-event page, will that link continue to work or return a 404?

**Comment by ph-fritsche** [comment]
Good point. For now the old page isn't removed. Later I would remove it from the navigation, but keep the page itself as long as there is a significant number of downloads for v13.

**Review by eps1lon — APPROVED** [review]

File: docs/ecosystem-user-event.mdx [hunk 2]
```diff
@@ -10,9 +10,10 @@ import TabItem from '@theme/TabItem'
 advanced simulation of browser interactions than the built-in
 [`fireEvent`](dom-testing-library/api-events.mdx#fireevent) method.
 
-> This page describes `user-event@13.5.0`.
-If you are starting or actively working on a project,
-we recommend to use [`user-event@14.0.0-beta`](user-event/intro.mdx) instead, as it includes important bug fixes and new features.  
+> This page describes `user-event@13.5.0`.
```
[the remark this answers is not in the document]
Comment (ph-fritsche) [reply docs/ecosystem-user-event.mdx]: I converted those blockquotes on the intro into admonitions. :)

**Comment by ph-fritsche** [comment]
v14 has been [released](https://github.com/testing-library/user-event/releases/tag/v14.0.0) :rocket:

**Pull request merged**